templates/zrobot/zrobot.py

```diff
@@ -0,0 +1,117 @@
+from js9 import j
+from zerorobot.template.base import TemplateBase
+from zerorobot.template.state import StateCheckError
+
+
+class Zrobot(TemplateBase):
+
+    version = '0.0.1'
+    template_name = "zrobot"
+
+    NODE_TEMPLATE = 'github.com/openvcloud/0-templates/node/0.0.1'
+    DOCKER_IMAGE = 'jumpscale/0-robot:telegram'
+
+    def __init__(self, name, guid=None, data=None):
+        super().__init__(name=name, guid=guid, data=data)
+
+        self._ovc = None
+        self._account = None
+        self._space = None
+
+    def validate(self):
+        for key in ['node', 'port']:
+            if not self.data[key]:
+                raise ValueError('%s is required' % key)
+
+        # validate accounts
+        nodes = self.api.services.find(template_uid=self.NODE_TEMPLATE, name=self.data['node'])
+
+        if len(nodes) != 1:
+            raise RuntimeError('found %s nodes, requires exactly one' % len(nodes))
+
+    def _prepare_repos(self, prefab, base):
+        for dir in ['data', 'config', 'ssh']:
+            prefab.core.dir_ensure(j.sal.fs.joinPaths(base, dir))
+
+        for dir in ['data', 'config']:
+            prefab.core.run('cd %s && git init' % j.sal.fs.joinPaths(base, dir))
+
+        key_dir = j.sal.fs.joinPaths(base, 'ssh')
+        if not prefab.core.exists('%s/id_rsa' % key_dir):
+            prefab.core.run('ssh-keygen -b 2048 -t rsa -f %s/id_rsa -q -N ""' % key_dir)
+
+    @property
+    def node(self):
+        nodes = self.api.services.find(template_uid=self.NODE_TEMPLATE, name=self.data['node'])
+
+        return nodes[0]
+
+    def install(self, force=False):
+        try:
+            self.state.check('actions', 'install', 'ok')
+            if not force:
+                return
+        except StateCheckError:
+            pass
+
+        node = j.tools.nodemgr.get(self.data['node'])
+        prefab = node.prefab
+        prefab.virtualization.docker.install()
+
+        prefab.core.run('docker rm -vf %s' % self.name, die=False)
```

Comment:
can't use the docker prefab module ? Here we do everything using the `core.run`, that should be abstracted in a prefab module